pkg/network/tracer/tracer_test.go

```diff
@@ -1001,11 +1007,14 @@ const (
 	validDNSServer = "8.8.8.8"
 )
 
-func testDNSStats(t *testing.T, domain string, success int, failure int, timeout int, serverIP string) {
-	config := testConfig()
-	config.CollectDNSStats = true
-	config.DNSTimeout = 1 * time.Second
-	tr := setupTracer(t, config)
+func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, timeout int, serverIP string) {
+	t.Cleanup(func() { tr.removeClient(clientID) })
+	t.Cleanup(func() { tr.ebpfTracer.Pause() })
+	require.NoError(t, tr.ebpfTracer.Pause(), "disable probes")
```

Comment:
I think you can remove the `Pause` and `Resume` for these tests because there isn't any work that needs to happen outside the "view" of the tracer. The resetting of the connections and DNS stats is what is important, which is handled by `removeClient`.